it/v1/endpoints/AmendPeriodSummaryControllerISpec.scala

```diff
@@ -200,9 +166,92 @@ class AmendPeriodSummaryControllerISpec extends IntegrationBaseSpec with JsonErr
           (INTERNAL_SERVER_ERROR, "SERVER_ERROR", INTERNAL_SERVER_ERROR, InternalError)
         )
 
-        input.foreach(args => (serviceErrorTest _).tupled(args))
+        val extraTysErrors = Seq(
+          (BAD_REQUEST, "INVALID_TAX_YEAR", BAD_REQUEST, TaxYearFormatError),
+          (UNPROCESSABLE_ENTITY, "TAX_YEAR_NOT_SUPPORTED", BAD_REQUEST, RuleTaxYearNotSupportedError),
+          (BAD_REQUEST, "INVALID_CORRELATION_ID", INTERNAL_SERVER_ERROR, InternalError),
+          (BAD_REQUEST, "INVALID_INCOMESOURCE_ID", BAD_REQUEST, BusinessIdFormatError),
+          (NOT_FOUND, "PERIOD_NOT_FOUND", NOT_FOUND, NotFoundError),
+          (NOT_FOUND, "INCOME_SOURCE_NOT_FOUND", NOT_FOUND, NotFoundError),
+          (NOT_FOUND, "INCOME_SOURCE_DATA_NOT_FOUND", NOT_FOUND, NotFoundError),
+          (UNPROCESSABLE_ENTITY, "BOTH_CONS_BREAKDOWN_EXPENSES_SUPPLIED", BAD_REQUEST, RuleBothExpensesSuppliedError)
+        )
+
+        (errors ++ extraTysErrors).foreach(args => (serviceErrorTest _).tupled(args))
       }
     }
   }
 
-}
+  private trait Test {
+
+    val nino: String       = "AA123456A"
+    val businessId: String = "XAIS12345678910"
+    val periodId: String   = "2019-01-01_2020-01-01"
+    val from               = "2019-01-01"
+    val to                 = "2020-01-01"
+
+    val responseJson: JsValue = Json.parse(
+      s"""
+         |{
+         |  "links": [
+         |    {
+         |      "href": "/individuals/business/self-employment/$nino/$businessId/period/$periodId",
+         |      "rel": "amend-self-employment-period-summary",
+         |      "method": "PUT"
+         |    },
+         |    {
+         |      "href": "/individuals/business/self-employment/$nino/$businessId/period/$periodId",
+         |      "rel": "self",
+         |      "method": "GET"
+         |    },
+         |    {
+         |      "href": "/individuals/business/self-employment/$nino/$businessId/period",
+         |      "rel": "list-self-employment-period-summaries",
+         |      "method": "GET"
+         |    }
+         |  ]
+         |}
+         |""".stripMargin)
+
+    def mtdUri: String = s"/$nino/$businessId/period/$periodId"
+    def downstreamUri: String
+    def setupStubs(): StubMapping
+
+    def request(): WSRequest = {
+      setupStubs()
+      buildRequest(mtdUri)
+        .withHttpHeaders(
+          (ACCEPT, "application/vnd.hmrc.1.0+json"),
+          (AUTHORIZATION, "Bearer 123")
+        )
+    }
+
+    def errorBody(code: String): String =
+      s"""
+         |      {
+         |        "code": "$code",
+         |        "reason": "message"
+         |      }
+    """.stripMargin
+
+  }
+
+    private trait TysIfsTest extends Test {
+
+      def downstreamTaxYear: String = "23-24"
+      def downstreamUri: String = s"/income-tax/$downstreamTaxYear/$nino/self-employments/$businessId/periodic-summaries"
+      def downstreamQueryParams: Map[String, String] = Map(
```

Comment:
`downstreamQueryParams` can probably be in `Test` as there's no difference between TYS and non-TYS.